### ISSUE TITLE
_conditional_probability return nan

### ISSUE THREAD

**Issue by ben0i0d** [alonfnt/tsnex#8]
code ：
```
import jax
import tsnex
import jax.numpy as jnp

key = jax.random.key(0)

x = jax.random.normal(key, shape=(1000, 50))
y = jnp.ones((1000,))

dist_fn = jax.vmap(jax.vmap(tsnex.tsne.euclidean_distance, in_axes=(0, None)), in_axes=(None, 0))

D = dist_fn(x,x)
sigma = tsnex.tsne._binary_search_perplexity(D,30.0)
p = tsnex.tsne._conditional_probability(D, sigma)
```
p is 
```
Array([[nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan],
       ...,
       [nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan]], dtype=float32)
```

When I tried to visualize the results, I found that the output generated by tsnex produced a blank image. 

To further investigate, I provided the above example to minimize and reproduce this issue.

My conclusion is that this is caused by the value of sigma being too small. 

In the given example, the sigma value is `6.42101e-20`. After performing the operation `2 * sigma**2`, this value underflows to 0.

This leads to an error in the calculation of `-distances / (2 * sigma**2)`.

![图片](https://github.com/user-attachments/assets/1231f97c-b9f6-471d-b8ac-8d3dac374fb8)

Since I am not very skilled in numerical algorithms, I submit this issue. If you have time, please take a look.

**Comment by alonfnt** [comment]
That is a great observation. I'll definitely try to fix it.

**Comment by ben0i0d** [comment]
Yes, I have verified that the issue has been fixed in version 0.0.3. Thank you. @alonfnt
